corehq/apps/sms/static/sms/js/phone_number_filter.js

```diff
@@ -42,11 +44,11 @@ hqDefine("sms/js/phone_number_filter", function() {
         });
     }
 
-    var initial_value = {{ initial_value| JSON }},
-        model = new SMSPhoneNumberFilterViewModel(initial_value, {{ groups| JSON }});
-    $('#{{ css_id }}').koApplyBindings(model);
+    var initialValue = initialPageData.get('initial_value'),
+        model = new SMSPhoneNumberFilterViewModel(initialValue, initialPageData.get('groups'));
+    $('#' + initialPageData.get("css_id")).koApplyBindings(model);
```

Comment:
Setting `css_id` up isn't a good pattern for these report filters, because there can be more than one of the same type of filter on the same page. It'll work in this case because this filter is only used in one report, and only used once in that report, but it'd be better to set this up as a widget, so there's a class like `report-filter-phone-number` and then initialize all widgets of that class in [reports/js/filters.js](https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/reports/static/reports/js/filters.js). Check out https://github.com/dimagi/commcare-hq/pull/17184 for a PR that does something similar.